src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -711,6 +715,22 @@ private void DowngradeLockFileIfNeeded(LockFile lockFile)
             }
         }
 
+        private void TouchMetadata(CacheFile cacheFile)
+        {
+            foreach (var packageFilePath in cacheFile.ExpectedPackageFilePaths)
+            {
+                var nupkgMetadataPath = Path.Combine(Path.GetDirectoryName(packageFilePath), ".nupkg.metadata");
```

Comment:
Wouldn't this end up touching fallback folders as well?

Reply:
A second unit test (or maybe modifying the new one in this PR) that validates that fallback folder .nupkg.metadata files can ensure this doesn't happen.

The test setup will be even more involved, but fallback folders use the same folder structure as the global packages folder, they're just supposed to be read-only. So, my best guess is that the test can restore a package into the global pacakges folder, move it to a fallback folder (create one if needed), modify the test's nuget.config to add this as a fallback folder, set the .nupkg.metadata file to an old timestamp, restore a second time, and ensure the file's timestamp has not changed.

Reply:
Should be fixed now.